src/autoscaler/eventgenerator/generator/appEvaluationManager.go

```diff
@@ -125,3 +133,9 @@ func (a *AppEvaluationManager) GetBreaker(appID string) *circuit.Breaker {
 	defer a.lock.Unlock()
 	return a.breakers[appID]
 }
+
+func (a *AppEvaluationManager) SetCoolDownExpired(appID string, expiredAt int64) {
+	a.lock.Lock()
```

Comment:
Is it better to provide two locks to setcooldown and circuitbreaker as they are independent?